compiler/rustc_data_structures/src/sync/worker_local.rs

```diff
@@ -42,7 +42,7 @@ pub struct Registry(Arc<RegistryData>);
 thread_local! {
     /// The registry associated with the thread.
     /// This allows the `WorkerLocal` type to clone the registry in its constructor.
-    static REGISTRY: OnceCell<Registry> = OnceCell::new();
+    static REGISTRY: OnceCell<Registry> = const { OnceCell::new() };
```

Comment:
What is the benefit of doing this?

Reply:
according to the docs it can be more efficient to wrap in `const`
https://rust-lang.github.io/rust-clippy/master/index.html#/thread_local_initializer_can_be_made_const

Reply:
A more practical demonstration: https://godbolt.org/z/K1Yxe8193 (You can see that the first variant does lazy initialization and the second does not).

Reply:
Gotcha, I missed that this is a thread_local! particular thing.